src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,49 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class type) {
+        List<T> list = new ArrayList<>();
+        if (Bulldozer.class == type) {
+            MachineProducer b = new BulldozerProducer();
+            list = b.get();
+            return new ArrayList<>(list);
```

Comment:
```suggestion
        if (Bulldozer.class == type) {
            return new ArrayList<>(bulldozerProducer.get());
```
lets create class-level variables for each producer and invoke it here - instead of creating a producer each time this method is invoked

ps. remember about oop principles when choosing type of variables